inc/Addon/ServiceProvider.php

```diff
@@ -66,6 +62,23 @@ public function register() {
 			->withArgument( $this->getContainer()->get( 'options' ) );
 
 		// Cloudflare Addon.
+		$this->addon_cloudflare();
+	}
+
+	/**
+	 * Adds Cloudflare Addon into the Container when the addon is enabled.
+	 *
+	 * @since 3.5
+	 */
+	protected function addon_cloudflare() {
+		// If the addon is not enabled, delete the transient and bail out. Don't load the addon.
+		if ( ! get_rocket_option( 'do_cloudflare' ) ) {
```

Comment:
Wouldn't it be faster to use the Options_Data instance that is in the container, instead of using `get_rocket_option` which will create a new instance?

Reply:
Yes, you're right. I'll change that.

Reply:
@Tabrisrp After looking at how we've got options wired into the container, it's not a singleton. Each time we do `$this->getContainer()->get( 'options' )`, it creates a new instance of `Options_Data`. The behavior is the same as using `get_rocket_option()` with the exception that it doesn't have to go through the container itself.

To test this for yourself, run this code:

```php
$container = apply_filters( 'rocket_container', '' );
var_dump( spl_object_id( $container->get( 'options' ) ) );
var_dump( spl_object_id( $container->get( 'options' ) ) );
```

You'll get a different object ID from each.

Therefore, in this case, `get_rocket_option()` will actually be a teeny tiny bit faster.

Reply:
Ah right, I forgot it was not returning the same instance each time on this. Does it make sense though? We don't really need a new instance each time we get this.

It's nitpicking at this point since it won't make a big difference. My feeling is that it feels better not mixing container code with some rocket specific code we have when we could have container code only there. This can be looked at later too.

Reply:
Ah right, I forgot it was not returning the same instance each time on this. Does it make sense though? We don't really need a new instance each time we get this.

It's nitpicking at this point since it won't make a big difference. My feeling is that it feels better not mixing container code with some rocket specific code we have when we could have container code only there. This can be looked at later too.